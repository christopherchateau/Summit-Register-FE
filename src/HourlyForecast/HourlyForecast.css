.HourlyForecast {
  align-items: center;
  background:white;
  border: 3px solid white;
  border-radius: 20px;
  color: black;
  display: flex;
  flex-direction: column;
  margin: 0 10px;
  min-height: 40vh;
  min-width: 200px;
  overflow-x: hidden;
  overflow-y: auto;
  padding: 5px;
  justify-content: space-evenly;
}

.weather-icon {
  height: 120px;
}

.HourlyForecast > h3 {
  font-size: 14px;
  margin: 2px;
}

.HourlyForecast > .time {
  border-bottom: 2px solid black;
  font-size: 20px;
}