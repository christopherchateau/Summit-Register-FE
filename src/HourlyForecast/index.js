import React, { Component } from "react";
import clearDay from "../utilities/Images/weather-icons/clear-day.svg";
import clearNight from "../utilities/Images/weather-icons/clear-night.svg";
import cloudy from "../utilities/Images/weather-icons/cloudy.svg";
import fog from "../utilities/Images/weather-icons/fog.svg";
import partlyCloudyDay from "../utilities/Images/weather-icons/partly-cloudy-day.svg";
import partlyCloudyNight from "../utilities/Images/weather-icons/partly-cloudy-night.svg";
import snow from "../utilities/Images/weather-icons/snow.svg";
import rain from "../utilities/Images/weather-icons/rain.svg";
import sleet from "../utilities/Images/weather-icons/sleet.svg";
import wind from "../utilities/Images/weather-icons/wind.svg";
import { convertUnixTimeStamp } from "../utilities/helper/timeStamp";
import PropTypes from "prop-types";
import "./HourlyForecast.css";

class HourlyForecast extends Component {
  covertToPercent = num => {
    return (num * 100).toFixed(0) + "%";
  };

  render() {
    const weatherIcons = {
      "clear-day": clearDay,
      "clear-night": clearNight,
      "partly-cloudy-night": partlyCloudyDay,
      "partly-cloudy-day": partlyCloudyNight,
      cloudy,
      rain,
      sleet,
      snow,
      wind,
      fog
    };

    const {
      time,
      icon,
      summary,
      humidity,
      windSpeed,
      temperature,
      precipProbability
    } = this.props;

    return (
      <div className="HourlyForecast">
        <h3 className="time">{convertUnixTimeStamp(time)}</h3>
        <img className="weather-icon" src={weatherIcons[icon]} alt={summary} />
        <h3>{summary}</h3>
        <h3>{temperature.toFixed(0)} Â°F</h3>
        <h3>Wind: {windSpeed.toFixed(0)} mph</h3>
        <h3>Chance of Rain: {this.covertToPercent(precipProbability)}</h3>
        <h3>Humidity: {this.covertToPercent(humidity)}</h3>
      </div>
    );
  }
}

HourlyForecast.propTypes = {};

export default HourlyForecast;
