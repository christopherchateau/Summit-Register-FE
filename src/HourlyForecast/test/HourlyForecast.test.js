import React from "react";
import ReactDOM from "react-dom";
import { shallow } from "enzyme";
import HourlyForecast from "..";

describe("HourlyForecast", () => {
  let wrapper;
  const currentMountainWeather = {
    time: 1546041600,
    summary: "Mostly Cloudy",
    icon: "clear-night",
    precipProbability: 0.55,
    temperature: 72.8,
    humidity: 0.95,
    windSpeed: 12.39
  };
  
  beforeEach(() => {
    wrapper = shallow(<HourlyForecast {...currentMountainWeather} />);
  });

  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<HourlyForecast {...currentMountainWeather} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
});
