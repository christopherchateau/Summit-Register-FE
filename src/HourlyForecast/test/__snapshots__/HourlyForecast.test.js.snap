// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HourlyForecast should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HourlyForecast
    humidity={0.95}
    icon="clear-night"
    precipProbability={0.55}
    summary="Mostly Cloudy"
    temperature={72.8}
    time={1546041600}
    windSpeed={12.39}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3
          className="time"
        >
          5pm
        </h3>,
        <img
          alt="Mostly Cloudy"
          className="weather-icon"
          src="clear-night.svg"
        />,
        <h3>
          Mostly Cloudy
        </h3>,
        <h3>
          73
           °F
        </h3>,
        <h3>
          Wind: 
          12
           mph
        </h3>,
        <h3>
          Chance of Rain: 
          55%
        </h3>,
        <h3>
          Humidity: 
          95%
        </h3>,
      ],
      "className": "HourlyForecast",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "5pm",
          "className": "time",
        },
        "ref": null,
        "rendered": "5pm",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "Mostly Cloudy",
          "className": "weather-icon",
          "src": "clear-night.svg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Mostly Cloudy",
        },
        "ref": null,
        "rendered": "Mostly Cloudy",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "73",
            " °F",
          ],
        },
        "ref": null,
        "rendered": Array [
          "73",
          " °F",
        ],
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Wind: ",
            "12",
            " mph",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Wind: ",
          "12",
          " mph",
        ],
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Chance of Rain: ",
            "55%",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Chance of Rain: ",
          "55%",
        ],
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Humidity: ",
            "95%",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Humidity: ",
          "95%",
        ],
        "type": "h3",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3
            className="time"
          >
            5pm
          </h3>,
          <img
            alt="Mostly Cloudy"
            className="weather-icon"
            src="clear-night.svg"
          />,
          <h3>
            Mostly Cloudy
          </h3>,
          <h3>
            73
             °F
          </h3>,
          <h3>
            Wind: 
            12
             mph
          </h3>,
          <h3>
            Chance of Rain: 
            55%
          </h3>,
          <h3>
            Humidity: 
            95%
          </h3>,
        ],
        "className": "HourlyForecast",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "5pm",
            "className": "time",
          },
          "ref": null,
          "rendered": "5pm",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "Mostly Cloudy",
            "className": "weather-icon",
            "src": "clear-night.svg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Mostly Cloudy",
          },
          "ref": null,
          "rendered": "Mostly Cloudy",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "73",
              " °F",
            ],
          },
          "ref": null,
          "rendered": Array [
            "73",
            " °F",
          ],
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Wind: ",
              "12",
              " mph",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Wind: ",
            "12",
            " mph",
          ],
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Chance of Rain: ",
              "55%",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Chance of Rain: ",
            "55%",
          ],
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Humidity: ",
              "95%",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Humidity: ",
            "95%",
          ],
          "type": "h3",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
